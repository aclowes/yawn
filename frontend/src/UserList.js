import React from 'react';
import {Table, Alert} from 'react-bootstrap';
import {Link} from 'react-router';

import API from "./API";

export default class UserList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {users: null, error: null};
  }

  componentDidMount() {
    API.get(`/api/users/`, (payload, error) => {
      this.setState({users: payload, error});
    });
  }

  renderRows() {
    if (this.state.users === null) {
      return (
        <tr>
          <td colSpan="3">Loading...</td>
        </tr>
      )
    } else {
      return this.state.users.map((user) => {
        return (
          <tr key={user.id}>
            <td><Link to={`/users/${user.id}`}>{user.username}</Link></td>
            <td>{user.email}</td>
            <td>{user.token}</td>
          </tr>
        )
      })
    }
  }

  render() {
    return (
      <div>
        <h3>Users</h3>
        {this.state.error && <Alert bsStyle="danger">{this.state.error}</Alert>}
        <Table striped bordered condensed hover>
          <thead>
          <tr>
            <th>Username</th>
            <th>Email</th>
            <th>Token</th>
          </tr>
          </thead>
          <tbody>
          {this.renderRows()}
          </tbody>
        </Table>
      </div>
    )
  }
}
