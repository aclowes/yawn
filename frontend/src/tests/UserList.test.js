import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import UserList from '../UserList';
import {mockAPI} from './mocks'

it('UserList loading', () => {
  mockAPI([null]);
  const component = ReactTestRenderer.create(<UserList />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('UserList failure', () => {
  mockAPI();
  const component = ReactTestRenderer.create(<UserList />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('UserList success', () => {
  mockAPI([[{
    "id": 4,
    "name": "alec-graybeard.local 36470",
    "status": "exited",
    "start_timestamp": "2017-01-30T04:13:47.966080Z",
    "last_heartbeat": "2017-01-30T04:15:10.845289Z"
  }]]);
  const component = ReactTestRenderer.create(<UserList />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
