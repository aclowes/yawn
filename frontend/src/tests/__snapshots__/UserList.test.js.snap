// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList failure 1`] = `
<div>
  <h3>
    Users
  </h3>
  <div
    className="alert alert-danger"
    role="alert"
  >
    some error message
  </div>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          Username
        </th>
        <th>
          Email
        </th>
        <th>
          Token
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colSpan="3"
        >
          Loading...
        </td>
      </tr>
    </tbody>
  </table>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="btn btn-default"
      disabled={false}
      type="button"
    >
      Add User
    </button>
  </a>
</div>
`;

exports[`UserList loading 1`] = `
<div>
  <h3>
    Users
  </h3>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          Username
        </th>
        <th>
          Email
        </th>
        <th>
          Token
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colSpan="3"
        >
          Loading...
        </td>
      </tr>
    </tbody>
  </table>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="btn btn-default"
      disabled={false}
      type="button"
    >
      Add User
    </button>
  </a>
</div>
`;

exports[`UserList success 1`] = `
<div>
  <h3>
    Users
  </h3>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          Username
        </th>
        <th>
          Email
        </th>
        <th>
          Token
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a
            onClick={[Function]}
            style={Object {}}
          >
            tester
          </a>
        </td>
        <td>
          bob@smith.com
        </td>
        <td>
          abcdef
        </td>
      </tr>
    </tbody>
  </table>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="btn btn-default"
      disabled={false}
      type="button"
    >
      Add User
    </button>
  </a>
</div>
`;
